extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -108,6 +110,18 @@ export class AmpGeo extends AMP.BaseElement {
     geoResolver(geo);
   }
 
+  /** @override */
+  layoutCallback() {
+    // Let the runtime know we just if we added/changed classes that
+    // may case a re-layout.
+    if (this.mutated_) {
+      return this.mutateElement(() => {
```

Comment:
This needs to be down where the body mutations happen.

Reply:
The issue I had is if I do it in the buildCallback then it fails because the element isn't attached ... layoutCallback was the earliest that it would work.

Reply:
the `getResources()` in `baseElement.measureMutateElement()` fails with "'no resources yet, since element is not attached"

Reply:
Stack please. It should definitely have connected by the time `#buildCallback` runs.

Reply:
Ah, found it, I wasn't attaching the element in my unit tests :-(   Amended code shortly.